packages/visualizations/src/components/MapPoi/utils.ts

```diff
@@ -0,0 +1,31 @@
+import type { BBox } from 'geojson';
+import type { PoiMapLayer } from './types';
+import type { Color } from '../types';
+import { DEFAULT_COLORS } from './constants';
+
+export const computeBaseRoundMarkerLayer = (
+    colors: Color[],
+    matchValues: string[],
+    matchKey: string,
+    noMatchColor?: Color
+): PoiMapLayer => {
+    const matchExpression = ['match', ['get', matchKey]];
+    matchValues.forEach((key, i) => {
+        matchExpression.push(key);
+        matchExpression.push(colors[i]);
+    });
```

Comment:
`matchValues` and `colors` seem to work in pairs. I'm concerned that the number of values in each of these arrays could be out of sync, causing the colors to shift by one.

Using a map between a value and a color prevent this. Futhermore it need one prop instead of two

from: 
```js
color: ['blue', 'green']
matchValues: ['City', 'Mountain']
matchKey: 'category'
```

to
```js
groupByKey: 'category';
groupByColors: {city: 'blue', montain: 'green' }
```